src/components/events/EditModal.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from "react";
+import axios from "axios";
+import { XIcon } from "../icons/Icons";
+import { ErrorText } from "../form/ErrorText";
+import { LoadingAnimation } from "../icons/LoadingAnimation";
+import { DAYS_ENUM } from "../../utils/common";
+import { GenericInput } from "../form/GenericInput";
+import { IdIcon } from "../icons/Icons";
+import { useUserState } from "../../utils/userState";
+import { DeleteDeal } from "../../server/deal";
+export default function EditModal({ eventData, onDismiss }) {
+    const [user] = useUserState();
+    const [loading, setLoading] = React.useState(false);
+    const { placeId,
+        dayOfWeek,
+        startTime,
+        endTime,
+        title,
+        deals,
+        place,
+        verifiedAt,
+        _id,
+        tags } = eventData
+    const [newDeal, setNewDeal] = useState({})
+    const [error, setError] = useState('')
+    const onClickEditDeal = async () => {
+        try {
+            if(!newDeal) {
+                setError('Please fill fields.')
+                return
+            }
+            await axios.put(`${process.env.REACT_APP_API_URL}/api/deal/edit/${_id}`, {
+                ...newDeal
+            }, {
+                headers: {
+                  Authorization: `${user.token}`,
+                },
+              })
+              window.location.reload()
+        } catch(e) {
+            setError(e)
+            console.log(e)
+        }
+    }
+
+    const onClickDelete = async () => {
+        try {
+            await DeleteDeal(_id, user);
+            window.location.reload()
+        } catch(e) {
+            console.log(e)
+        }
+    }
+    return (
+        <div className="fixed top-0 left-0 w-full h-full z-50">
+            <div
+                className="absolute top-0 left-0 h-full w-full bg-gray-500 opacity-50 z-30 overflow-hidden"
+                onClick={() => onDismiss()}
+            />
+            {/* <!-- Modal content --> */}
+            <div className="flex justify-center ">
+                <div
+                    id="reportModal"
+                    tabIndex="-1"
+                    aria-hidden="true"
+                    className={`fixed top-1/4 mx-auto h-fit z-50 flex w-fit p-4 overflow-hidden`}
```

Comment:
nit: this should be a "fixed top-1/4 mx-auto h-fit z-50 flex w-fit p-4 overflow-hidden" rather that an object {``}